### ISSUE TITLE
Ingest Yale Papyrus Collection to DCS

### ISSUE THREAD

**Issue by sshetenhelm** [yalelibrary/YUL-DC#2121]
**Story**
The Yale Papyrus Collection (PID 42) is currently in FindIt and needs to move to DCS. However, will use this collection as a test of the new Preservica DCS integration. Will ingest the collection into Preservica Test and DCS UAT first. Then when Preservica V6 is in Prod, will ingest final collection there. BRBL have said that Original parent OIDs DON'T need to be retained from Findit, so this is not longer a migration, but instead a new collection ingest using Preservica

**Acceptance**

- [x] Talk to BRBL, determine if original parent OIDs need to be retained or if we can use this collection for Preservica testing 
Edit: 2022.06.03 Original parent OIDs DON'T need to be retained
- [x] DPS and BRBL to add metadata to ASpace TEST
- [x] DPS and BRBL to resync content in Preservica TEST
- [x] Ingest collection into DCS UAT
- [x] Troubleshoot problem objects
- [x] Add content to Preservica Production and ASpace Prod
- [ ] Ingest into DCS production
- [x] Ask BRBL if they want old Findit handles redirected to DCS **(EDIT: YES)**
- [ ] Update old handles to redirect to FindIt

**Comment by motropuk** [comment]
Have written to BRBL to see if the Papryi data in Preservica Test is complete https://yul-pres-tsdb.library.yale.edu/explorer/explorer.html#browse:SO&ecd02c4b-21a9-4628-80e5-f257ede28f97&null and whether the ASpace handles are now in Test and or Prod

**Comment by motropuk** [comment]
David and Mark are working on getting this collection into ASpace TEST and Preservica TEST so we can add it to DCS UAT.

**Comment by motropuk** [comment]
@sshetenhelm did we want to pull this into Sprint at some point, as this work is now ongoing?

**Comment by sshetenhelm** [comment]
Yes, I can go ahead and slide it in.

**Comment by sshetenhelm** [comment]
@ucancallmealicia is the papyri metadata in test Aspace? Or is that still in progress?

**Comment by ucancallmealicia** [comment]
Yes it's in test in the BRBL repo: https://testarchivesspace.library.yale.edu/resources/5894

**Comment by sshetenhelm** [comment]
DPS is still doing testing in Preservica re: the collection hierarchy.

**Comment by sshetenhelm** [comment]
TEST resync is complete.

**Comment by sshetenhelm** [comment]
@ucancallmealicia Aspace is saying YPC has an unpublished ancestor. Is that why I can't seem to find it on test Archives at Yale?

**Comment by ucancallmealicia** [comment]
@sshetenhelm the resource record was unpublished, so yes that's why you were getting that message. I just published it so should show up in a few.

**Comment by sshetenhelm** [comment]
Currently adding Fixity to entire collection in order to ingest in bulk.

**Comment by sshetenhelm** [comment]
Using information from ArchivesSpace export, created 6894 parents in MGMT UAT. 
23 parents did not pull children -- will investigate
PID in Ladybird only has 6462 parents, while FindIt shows 6553 total parents. 

@motropuk and @ucancallmealicia, do either of you know a definitive amount of parent objects we should have?

**Comment by sshetenhelm** [comment]
For objects that had zero children in MGMT, folder in Preservica appeared to have no images. Still investigating.

**Comment by sshetenhelm** [comment]
Based on spreadsheet papyri-2018-excel-master, was able to match and remove all parents attached to records of objects without scans.

Now have 6319 parents in Management, which is still a discrepancy with Ladybird PID 42 (6462 - 6319 = 143). Possible some LB records hierarchical? Will continue investigating.

**Comment by sshetenhelm** [comment]
141 objects are in FindIt but do not have TIFFs, as per legacy spreadsheet. These objects are:
[YPC_NoTIFFs.csv](https://github.com/yalelibrary/YUL-DC/files/13832399/YPC_NoTIFFs.csv)

11 objects are in FindIt but did not bring children into DCS, but should have TIFFs, as per legacy spreadsheet. These objects are:
[YPC_MissingScans.csv](https://github.com/yalelibrary/YUL-DC/files/13832407/YPC_MissingScans.csv)

**Comment by sshetenhelm** [comment]
Ingested 9 of 11 outstanding objects, thanks David for helping us locate! 

In Management - 6,328 parents 
In Blacklight - 6,328 parents

Still investigating P.CtYBR inv. 4717(B) & P.CtYBR inv. 3679(B)
------

6914 Records on Mark Custer’s Spreadsheet
587 should have zero images

6914 - 587 = 6327

6328 in MGMT + 2 outstanding = 6331

Not sure how we are four over, but I will investigate.

**Comment by sshetenhelm** [comment]
Identified 5 duplicates of parents in UAT (same aspace URI, same Preservica info). Deleted.
Now 6,323 parents in MGMT UAT.

6,323 + 2 still under investigation = 6,325

Now we are two objects under legacy list

Will continue to investigate

**Comment by sshetenhelm** [comment]
- Alicia has confirmed she can start moving things around in PROD Aspace.
- Successfully ingested P.CtYBR inv. 358(B)
- Tried to ingest P.CtYBR inv. 4311. Second image may have problem in Preservica. Have asked DPS to investigate.
- Two objects need digital files in Preservica. As per legacy spreadsheet, their file names should be as follows: 

-- P.CtYBR inv. 4717(B) - PCtYBR_inv_4717_vo.tif
-- P.CtYBR inv. 3679(B) - 1022316.tif

**Comment by sshetenhelm** [comment]
Update from DPS - Aspace link workflow is working, however, items not moving automatically to BRBL Preservica folder. Putting down temporarily while people are out of office, will pick back up week of 19th.

**Comment by sshetenhelm** [comment]
Pivoting back to this. Records are being published to Aspace as we speak, will attempt first ingest this afternoon.

**Comment by sshetenhelm** [comment]
Current [3,367 available in Blacklight PROD](https://collections.library.yale.edu/catalog?op=AND&sort=score+desc%2C+pub_date_si+desc%2C+title_ssim+asc%2C+archivalSort_ssi+asc&search_field=advanced&all_fields_advanced=&creator_tesim=&title_tesim=&callNumber_tesim=YPC&date_fields=&subject_fields=&genre_fields=&digitization_note_tesi=&fulltext_tsim_advanced=&oid_ssi=&child_oids_ssim=&commit=SEARCH).

**Comment by sshetenhelm** [comment]
Current stats:
5070 in Blacklight
5646 in MGMT

stilll working

**Comment by sshetenhelm** [comment]
Up to 5922 in Blacklight, still working through the last few hundred

**Comment by sshetenhelm** [comment]
Goal is 6,323 in Blacklight
Currently 6,136
Difference of 187 objects

Still investigating.

**Comment by sshetenhelm** [comment]
Currently 6144 objects in Blacklight.
Identified 104 objects that are in Preservica/Aspace TEST but not in Preservica/Aspace PROD. Working on reconciling.

**Comment by sshetenhelm** [comment]
6,253 in Blacklight

36 objects:
* Show as complete in Batch Process
* Have successfully pulled in Metadata from Aspace
* Confirmed to have TIF files in Preservica
* But contain 0 children regardless of how many re-syncs are attempted
CSV HERE: [YPC-0Children-Broken.csv](https://github.com/yalelibrary/YUL-DC/files/15152864/YPC-0Children-Broken.csv)
 
31 objects:
* Show as failed and parent deleted or pending in Batch Process
* Have metadata that is viewable using the Metadata Cloud uri, but refuses to pull into Management
* Confirmed to have TIF files in Preservica
* Contain no children and refuse to create child-level records
* Display the following error in the batch process:
`Request error 503 <?xml version="1.0" encoding="UTF-8" standalone="yes"?><Error><ExtendedMessage>Content does not exist in the storage.</ExtendedMessage><MessageKey>error.from.storage</MessageKey></Error>	`
CSV HERE: [YPC-Failed-Broken.csv](https://github.com/yalelibrary/YUL-DC/files/15152871/YPC-Failed-Broken.csv)

3 objects:
* Appear to have unpublished metadata records. Will see if there's a reason and if not, will publish and try again.

**Comment by sshetenhelm** [comment]
For the first three I tried in YPC-Failed-Broken, when I tried to right-click > Download on these objects, I received the following error message for at least one TIF in each folder: 
```
This page isn’t working preservica.library.yale.edu is currently unable to handle this request.
HTTP ERROR 500
```

The three YPC-0Children-Broken objects I tried all worked though.

**Comment by sshetenhelm** [comment]
YPC-Failed-Broken needed a fix in Preservica; our test worked, so just waiting for the green light to ingest the rest of that batch.

For YPC-0Children-Broken, From David:

_so far the first 4-5 that I've checked from YPC-0Children-Broken.csv don't have any download events from the s_dcs_brbl account in their histories,  so for those it looks like the request isn't making it to the download level_

**Comment by decirella** [comment]
@sshetenhelm the items on YPC-Failed-Broken are ready to be re-tried. 

The cause of these failures were stale file handles being presented by the storage product to Preservica, we were able to manually refresh these stale file handles. It's not clear yet why these specific files were afflicted, we are working towards a general solution.

**Comment by sshetenhelm** [comment]
YPC-Failed-Broken worked, but had to resync a few to get them going. 
Now 6,284 in PROD.

**Comment by sshetenhelm** [comment]
Successfully ingested the unpublished Aspace parents. 
Now 6,288 in Prod

The only outstanding objects are the 36 that refuse to create child objects. Will continue to investigate.

**Comment by sshetenhelm** [comment]
Case Study: Parent OID [33203833](https://collections.library.yale.edu/management/parent_objects/33203833)

- Ingested with [Batch Process 16997](https://collections.library.yale.edu/management/batch_processes/16977) 
- 0 child objects created
- Preservica re-sync does not create any children
- Successfully pulls in metadata from ArchivesSpace
- Preservica object looks similar to successfully uploaded Preservica objects, include similar XIP and Properties visible to users on Properties tag (correct Security Tag, object structure type, etc.)
- Object with same information was successfully uploaded into Preservica UAT at https://collections-uat.library.yale.edu/management/parent_objects/901604533 

Jobs for object: 
![Image](https://github.com/yalelibrary/YUL-DC/assets/98423494/496fd592-c3e2-49be-86f0-a22e3478ef8d)

Is there any way to look up Management Production jobs 13993956 and 13993957 to see if something went wonky there?

**Comment by sshetenhelm** [comment]
Did one more test today, having same issue. Created #2826 to address remaining 36 objects and will close this ticket. Problem described in previous comment is available in #2824 for troubleshooting purposes.
